pandas/indexes/interval.py

```diff
@@ -0,0 +1,750 @@
+        result._closed = closed
+        result._mask = mask
+        result.name = name
+        if not fastpath:
+            result._validate()
+        result._reset_identity()
+        return result
+
+    @Appender(_index_shared_docs['_shallow_copy'])
+    def _shallow_copy(self, left=None, right=None, **kwargs):
+        if left is None:
+            left = self.left
+            right = self.right
+
+        attributes = self._get_attributes_dict()
+        attributes.update(kwargs)
+        return self._simple_new(left, right, **attributes)
+
+    def _validate(self):
+        """Verify that the IntervalIndex is valid.
+        """
+        # TODO: exclude periods?
+        if self.closed not in _VALID_CLOSED:
+            raise ValueError("invalid options for 'closed': %s" % self.closed)
+        if len(self.left) != len(self.right):
+            raise ValueError('left and right must have the same length')
+        left_valid = notnull(self.left)
+        right_valid = notnull(self.right)
+        if not (left_valid == right_valid).all():
+            raise ValueError('missing values must be missing in the same '
+                             'location both left and right sides')
+        if not (self.left[left_valid] <= self.right[left_valid]).all():
+            raise ValueError('left side of interval must be <= right side')
+
+    @cache_readonly
+    def hasnans(self):
+        """ return if I have any nans; enables various perf speedups """
+        if self._mask is None:
+            return False
+        return self._mask.any()
+
+    @cache_readonly
+    def _isnan(self):
+        """ return if each value is nan"""
+        if self._mask is None:
+            self._mask = np.zeros(len(self), dtype=np.bool_)
+        return self._mask
+
+    @property
+    def _engine(self):
+        raise NotImplementedError
+
+    def _cleanup(self):
+        pass
+
+    @cache_readonly
+    def _tree(self):
+        return IntervalTree(self.left, self.right, closed=self.closed)
+
+    @property
+    def _constructor(self):
+        return type(self).from_intervals
+
+    @classmethod
+    def from_breaks(cls, breaks, closed='right', name=None):
+        """
+        Construct an IntervalIndex from an array of splits
+
+        Parameters
+        ----------
+        breaks : array-like (1-dimensional)
+            Left and right bounds for each interval.
+        closed : {'left', 'right', 'both', 'neither'}, optional
+            Whether the intervals are closed on the left-side, right-side, both
+            or neither. Defaults to 'right'.
+        name : object, optional
+            Name to be stored in the index.
+
+        Examples
+        --------
+
+        >>> IntervalIndex.from_breaks([0, 1, 2, 3])
+        IntervalIndex(left=[0, 1, 2],
+                      right=[1, 2, 3],
+                      closed='right')
+        """
+        breaks = np.asarray(breaks)
+        mask = isnull(breaks[:-1])
```

Comment:
I guess this is reasonable, but note that the only valid IntervalIndex you can construct with `from_breaks` that contains nulls is all null.